File: goberkeley/src/berkeley/logger_manager.go
package berkeley

import (
	"github.com/sirupsen/logrus"
	"reflect"
)

// LoggerManager es responsable de gestionar los loggers para las clases que lo invocan.
type LoggerManager struct{}

// GetLogger obtiene el logger correspondiente a la clase que invoca este método.
func (LoggerManager) GetLogger(clazz interface{}) *logrus.Logger {
	// Se usa el nombre de la clase (tipo) como parte del nombre del logger.
	logger := logrus.New()

	// Se asigna el nombre del tipo (nombre de la clase)
	logger.SetFormatter(&logrus.TextFormatter{
		FullTimestamp: true,
	})
	logger.SetLevel(logrus.InfoLevel)

	className := reflect.TypeOf(clazz).Name()
	logger.Infof("Logger for %s initialized.", className)

	return logger
}

File: jberkeley/src/org/jogugi/berkeley/LoggerManager.java
package org.jogugi.berkeley;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * La clase {@code LoggerManager} es responsable de gestionar los loggers
 * para las clases que lo invocan. Utiliza la biblioteca {@link slf4j} para
 * crear instancias de loggers.
 * <p>
 * Esta clase proporciona un único método estático que devuelve el logger
 * correspondiente a la clase que lo invoca.
 * </p>
 * 
 * @author Tu Nombre
 * @version 1.0
 * @since 2024-12-07
 */
public class LoggerManager {

    /**
     * Obtiene el logger correspondiente a la clase que invoca este método.
     * 
     * @param clazz La clase para la cual se desea obtener el logger.
     * @return Un objeto {@link Logger} correspondiente a la clase {@code clazz}.
     * @see Logger
     */
    public static Logger getLogger(Class<?> clazz) {
        return LoggerFactory.getLogger(clazz);
    }
}
